app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+require 'data_mapper'
+require 'dm-postgres-adapter'
+
+class User
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :email, String, required: true
+  property :password, String, required: true
+
+  validates_presence_of :email
+  validates_uniqueness_of :email
```

Comment:
If you have `required: true` in the property, you don't need validates_presence_of below
Also you can remove validates_uniqueness_off if you add `unique: true` to the property